### PR TITLE
Move namespace declarations to top to prevent ordering issues

### PR DESCRIPTION
**Pull request by lostthetrail**
This resolves issue reported in issue #127.

CLA found under corporate license HomeAway.com Inc.

### CODE REVIEW

**Comment by lazd** [comment]
@tkellen @shama @vladikoff can we get a review on this? The issue looks pretty important. I took a quick look and it makes sense, but I don't have time for a deep dive on this unfortunately.

**Comment by tkellen** [comment]
@lostthetrail, do you have an npm account? I just added you as a collaborator to this repo. I'd be glad to give you publish access as well if you do.  Also, @lazd, I've done the same for you.  I don't have the bandwidth to deal with this right now but I appreciate the PR and I hope that one of you can see it through!

**Comment by lostthetrail** [comment]
@tkellen I just created an npm account just now. Under the same moniker.

**Comment by lostthetrail** [comment]
@lazd As I put in the PR, would you mind reviewing this? If you don't have the time, I will play both parts.

**Comment by lazd** [comment]
@lostthetrail I'll take a look.

**Comment by tkellen** [comment]
@lostthetrail / @lazd you're both npm author's now.  please remember to update the changelog (it's yaml), run grunt after (to generate the readme), and tag your release (git tag -m "version").

**Comment by lazd** [comment]
@tkellen thanks, will do!

File: README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# grunt-contrib-handlebars v0.9.2 [![Build Status: Linux](https://travis-ci.org/gruntjs/grunt-contrib-handlebars.svg?branch=master)](https://travis-ci.org/gruntjs/grunt-contrib-handlebars) [![Build Status: Windows](https://ci.appveyor.com/api/projects/status/byxsu7xtyjxuwe3g/branch/master?svg=true)](https://ci.appveyor.com/project/gruntjs/grunt-contrib-handlebars/branch/master)
+# grunt-contrib-handlebars v0.9.2 [![Build Status: Linux](https://travis-ci.org/gruntjs/grunt-contrib-handlebars.svg?branch=master)](https://travis-ci.org/gruntjs/grunt-contrib-handlebars) [![Build Status: Windows]([object Object])](https://ci.appveyor.com/project/gruntjs/grunt-contrib-handlebars/branch/master)
```
Comment (lazd) [inline README.md]: Not sure what happened here, but this URL is wrong.

**Comment by lazd** [comment]
@lostthetrail I just took a look at the code and the tests/fixtures, this looks right. The original code wasn't taking into account the `partialsUseNamespace` option, and it's pretty clear to see that it was trying to assign partials to properties of namespaces before they were declared.

@lostthetrail, that one problem with the README is the only thing left. I'll go ahead and merge this tomorrow and fix that problem before releasing.

**Comment by lostthetrail** [comment]
Thanks @lazd !

**Pull request merged**